edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/RegistryController.cs

```diff
@@ -0,0 +1,747 @@
+                Events.CompleteRequest(nameof(this.DeleteModuleOnBehalfOfAsync), edgeDeviceId, targetId, requestData.AuthChain);
+            }
+            catch (Exception ex)
+            {
+                Events.InternalServerError(nameof(this.DeleteModuleOnBehalfOfAsync), ex);
+                await this.SendResponseAsync(HttpStatusCode.InternalServerError, this.FormatErrorResponseMessage(ex));
+            }
+        }
+
+        async Task<(bool, string)> AuthorizeActorAsync(IDeviceScopeIdentitiesCache identitiesCache, string actorDeviceId, string actorModuleId, string targetId)
+        {
+            if (actorModuleId != Constants.EdgeHubModuleId)
+            {
+                // Only child EdgeHubs are allowed to act OnBehalfOf of devices/modules.
+                Events.AuthorizationFail_BadActor(actorDeviceId, actorModuleId, targetId);
+                return (false, string.Empty);
+            }
+
+            // Actor device is claiming to be our child, and that the target device is its child.
+            // So we should have an authchain already cached for the target device.
+            Option<string> targetAuthChainOption = await identitiesCache.GetAuthChain(targetId);
+
+            if (!targetAuthChainOption.HasValue)
+            {
+                Events.AuthorizationFail_NoAuthChain(targetId);
+                return (false, string.Empty);
+            }
+
+            // Validate the target auth-chain
+            string targetAuthChain = targetAuthChainOption.Expect(() => new InvalidOperationException());
+            if (!AuthChainHelpers.ValidateAuthChain(actorDeviceId, targetId, targetAuthChain))
+            {
+                Events.AuthorizationFail_InvalidAuthChain(actorDeviceId, targetId, targetAuthChain);
+                return (false, string.Empty);
+            }
+
+            return (true, targetAuthChain);
+        }
+
+        async Task SendResponseAsync(HttpResponseMessage responseMessage)
+        {
+            string content = string.Empty;
+            if (responseMessage.Content != null)
+            {
+                content = await responseMessage.Content.ReadAsStringAsync();
+            }
+
+            Events.LogDebug($"SendResponseAsync: {responseMessage.StatusCode}, content={content}");
+            await this.SendResponseAsync(responseMessage.StatusCode, content);
+        }
+
+        async Task SendResponseAsync(HttpStatusCode status, string jsonContent = "")
+        {
+            this.Response.StatusCode = (int)status;
+
+            if (!string.IsNullOrEmpty(jsonContent))
+            {
+                var resultUtf8Bytes = Encoding.UTF8.GetBytes(jsonContent);
+
+                this.Response.ContentLength = resultUtf8Bytes.Length;
+                this.Response.ContentType = ContentType;
+
+                await this.Response.Body.WriteAsync(resultUtf8Bytes, 0, resultUtf8Bytes.Length);
+            }
+        }
+
+        string FormatErrorResponseMessage(Exception ex)
```

Comment:
Nit: These kind of single line methods are not very useful. I would suggest removing them. Or maybe keep just the last FormatJsonMessage

Reply:
Cleaned up.